esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2200,6 +2200,82 @@ inline const char *controlCmdMessage(int cmd)
     return NULL;
 }
 
+bool CWsSMCEx::onRoxieXrefCmd(IEspContext &context, IEspRoxieXrefCmdRequest &req, IEspRoxieXrefCmdResponse &resp)
+{
+    context.ensureFeatureAccess(ROXIE_CONTROL_URL, SecAccess_Full, ECLWATCH_SMC_ACCESS_DENIED, SMC_ACCESS_DENIED);
+
+    StringBuffer controlReq;
```

Comment:
design: @afishbeck does this solve the issue outline in the jira.  

> "In addition can we also have the '<control:getQueryXrefInfo/>'  "testsocket" functionality in ecl command line so we don't have to rely on roxie being up to get the queries/superkeys list"

I thought the intent was to retrieve this information from the data stored in dali, instead of going out to the roxie to retrieve it.

Reply:
Right.  After discussion with him offline I added the xref passthrough for cases where he can't get to the roxie via testsocket.
"ecl queries files" is meant to be the function that parallels ```<control:getQueryXrefInfo/>``` without going to roxie, but there are still a few enhancements that could be made that would make it more directly comparable.  

For example, 
1. right now it only works one query at a time.. but xref supports "all queries".  
2. It's quite a bit more work, but we could start keep track and showing the full nesting of superfiles.  I'm not sure what  ```<control:getQueryXrefInfo/>``` does there.

Perhaps I should open another JIRA for additional enhancements to "ecl queries files"?  I think these changes will help him right away.